keras/engine/topology.py

```diff
@@ -284,10 +284,14 @@ def __init__(self, **kwargs):
 
         # these properties will be set upon call of self.build(),
         # which itself will be called upon self.add_inbound_node if necessary.
-        self.trainable_weights = []
-        self.non_trainable_weights = []
-        self.regularizers = []
-        self.constraints = {}  # dict {tensor: constraint instance}
+        if not hasattr(self, 'trainable_weights'):
+                self.trainable_weights = []
+        if not hasattr(self, 'non_trainable_weights'):
+                self.non_trainable_weights = []
+        if not hasattr(self, 'regularizers'):
+                self.regularizers = []
+        if not hasattr(self, 'constraints'):
+                self.constraints = {}  # dict {tensor: constraint instance}
```

Comment:
So when these properties are set, the lists should be split to 2 and set to forward and backward rnns separately?
For e.g, what happens when I do bi_rnn.trainable_weights=[a,b,c,d]? Set [a,b] to forward rnn and [c, d] to backward rnn?
In that case what about constrains? Its a dict, the order will be messed up.